geomstats/geometry/connection.py

```diff
@@ -178,6 +178,7 @@ def objective(velocity):
 
         tangent_vec = gs.array(res.x)
         tangent_vec = gs.reshape(tangent_vec, max_shape)
+        tangent_vec = gs.cast(tangent_vec, dtype=base_point.dtype)
```

Comment:
NIT: why using the dtype from base_point, and not from point for example?

On a higher level, this could be related to issue #775

Reply:
Because tangent_vec is tangent at base point, and they might be used together in .e.g norm (for a distance computation)